zerver/tests/test_events.py

```diff
@@ -2204,7 +2204,8 @@ def do_set_user_display_settings_test(self, setting_name: str) -> None:
 
     def test_set_user_display_settings(self) -> None:
         for prop in UserProfile.property_types:
-            self.do_set_user_display_settings_test(prop)
+            if prop not in UserProfile.notification_settings_legacy:  # These have tests separately.
+                self.do_set_user_display_settings_test(prop)
```

Comment:
Should we rename this and [`do_set_user_display_setting`](https://github.com/zulip/zulip/blob/master/zerver/lib/actions.py#L5077) to something like `do_set_user_setting`?

Reply:
I agree we should rename this.

Reply:
Okay, will add a new commit tomorrow.

Reply:
Done. Changed it `do_set_user_setting` but it's confusing to not include notification settings in that. I've talked about an issue with moving the `do_change_notification_settings` code to this function [here](https://github.com/zulip/zulip/pull/19561#discussion_r688568744). Any ideas/thoughts appreciated! :)